### ISSUE TITLE
Bridge proxy using vmess

### ISSUE THREAD

**Issue by Pouriyahe** [v2ray/discussion#433]
This might seem like a noob thing but please help us.
Iran's government has physically cut internet access.
Here is the situation :

I (I am going to refer myself as the client) don't have access to internet.
I have a server (server A) that has access to internet and the client can connect to it via v2ray.
Now the real issue is that server A itself has a very restricted internet access (ie. Twitter is blocked on it's internet)
Now I have another server located in USA (which has very open internet)  say server B 
and I want server A to be take all the data from the client and send it to server B for it to send to the open internet.

In a nutshell I want access to server B's internet but I only have access to server A and server A has access to both me and server B.
Server A can be the link between me and server B.

I know some things about inbound but I'm a noob.
Both servers run Ubuntu
I have to use Vmess

Thanks for helping us !

**Comment by Pouriyahe** [comment]
How can this be done?
Can you show me an example for the vmess config that I can use?

**Comment by MN2yFii** [comment]
You can chain your servers like this.
client [ inbound: socks or http, outbound: vmess.A ] -> server A [ inbound: vmess.A, outbound: vmess.B ] -> server B [ inbound: vmess.B, outbound: freedom ]

Here are 3 config examples. Please test and debug them on your local machine, before deploy to the servers.
#### client (connect to server A)
```json
{
  "log": {
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "tag": "agentin",
      "port": 1080,
      "listen": "127.0.0.1",
      "protocol": "socks",
      "settings": {}
    }
  ],
  "outbounds": [
    {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "127.0.0.1",
            "port": 1081,
            "users": [
              {
                "id": "60935889-b531-4e65-80b2-f728c64c8835"
              }
            ]
          }
        ]
      },
      "tag": "agentout",
      "streamSettings": {
        "network": "ws",
        "security": "none",
        "wsSettings": {
          "path": "/serva"
        }
      }
    }
  ]
}
```

#### Server A (relay)
```json
{
  "log": {
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "tag": "agentin",
      "port": 1081,
      "listen": "127.0.0.1",
      "protocol": "vmess",
      "settings": {
        "clients": [
          {
            "id": "60935889-b531-4e65-80b2-f728c64c8835"
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "none",
        "wsSettings": {
          "path": "/serva"
        }
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "127.0.0.1",
            "port": 1082,
            "users": [
              {
                "id": "d6c97b8e-354d-4c3a-9541-247818eac979"
              }
            ]
          }
        ]
      },
      "tag": "agentout",
      "streamSettings": {
        "network": "ws",
        "security": "none",
        "wsSettings": {
          "path": "/servb"
        }
      }
    }
  ]
}
```

#### Server B
```json
{
  "log": {
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "tag": "agentin",
      "port": 1082,
      "listen": "127.0.0.1",
      "protocol": "vmess",
      "settings": {
        "clients": [
          {
            "id": "d6c97b8e-354d-4c3a-9541-247818eac979"
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "none",
        "wsSettings": {
          "path": "/servb"
        }
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {},
      "tag": "agentout",
      "streamSettings": {}
    }
  ]
}
```

For more information, please take a look at https://v2ray.com

**Comment by Pouriyahe** [comment]
@MN2yFii thanks for the responce
I tried and the client side and server B are ok and set up but there is a problem with server A(relay)
when starting v2ray I get : 
`v2ray.service - V2Ray Service
   Loaded: loaded (/etc/systemd/system/v2ray.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2019-11-21 11:26:39 UTC; 6s ago
  Process: 9289 ExecStart=/usr/bin/v2ray/v2ray -config /etc/v2ray/config.json (code=exited, status=23)
 Main PID: 9289 (code=exited, status=23)

Nov 21 11:26:39 ubuntu-standard1-at1 systemd[1]: Stopped V2Ray Service.
Nov 21 11:26:39 ubuntu-standard1-at1 systemd[1]: Started V2Ray Service.
Nov 21 11:26:39 ubuntu-standard1-at1 v2ray[9289]: V2Ray 4.21.3 (V2Fly, a community-driven edition of V2Ray.) Custom
Nov 21 11:26:39 ubuntu-standard1-at1 v2ray[9289]: A unified platform for anti-censorship.
Nov 21 11:26:39 ubuntu-standard1-at1 v2ray[9289]: main: failed to read config file: /etc/v2ray/config.json > v2ray.com/core/main/json: failed to execute v2ctl to convert config
Nov 21 11:26:39 ubuntu-standard1-at1 v2ray[9289]:  > exit status 255
Nov 21 11:26:39 ubuntu-standard1-at1 systemd[1]: v2ray.service: Main process exited, code=exited, status=23/n/a
Nov 21 11:26:39 ubuntu-standard1-at1 systemd[1]: v2ray.service: Failed with result 'exit-code'.
`
my config on server A is something like this : 
` {
    "log": {
        "access": "/var/log/v2ray/access.log",
        "error": "/var/log/v2ray/error.log",
        "loglevel": "warning"
    },
    "inbound": {
        "port": 12345,
        "protocol": "vmess",
        "settings": {
            "clients": [
                {
                    "id": "82db1f0d-8c40-58d5-8e3b-fd3f18fdc1f9",
                    "level": 1,
                    "alterId": 100
                }
            ]
        },

 "outbound":     {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "//Server B IP",
            "port": //Server B vmess port,
            "users": [
              {
                "id": "//one of serve B's clients",
"alterId": 100,
"security": "auto",                   
"level": 1
                    
              }
            ]
          }
        ]
      },
      "tag": "agentout",
      "streamSettings": {
        "network": "tcp",
        "security": "auto",
      }
    },

    "inboundDetour": [ ],
    "outboundDetour": [
        {
            "protocol": "blackhole",
            "settings": {},
            "tag": "blocked"
        }
    ],
    "routing": {
        "strategy": "rules",
        "settings": {
            "rules": [
                {
                    "type": "field",
                    "ip": [
                        "0.0.0.0/8",
                        "10.0.0.0/8",
                        "100.64.0.0/10",
                        "127.0.0.0/8",
                        "169.254.0.0/16",
                        "172.16.0.0/12",
                        "192.0.0.0/24",
                        "192.0.2.0/24",
                        "192.168.0.0/16",
                        "198.18.0.0/15",
                        "198.51.100.0/24",
                        "203.0.113.0/24",
                        "::1/128",
                        "fc00::/7",
                        "fe80::/10"
                    ],
                    "outboundTag": "blocked"
                }
            ]
        }
    }
}`

the config on server B is pretty similar
I used config generator and changed the outbound section to what you suggested

**Comment by MN2yFii** [comment]
The "inbound" section miss a trailing brace.
You can use online json validator to avoid this kind of error.

**Comment by Pouriyahe** [comment]
that's right
my bad
I'm kind of a noob sorry for that
just wanted to let you know the problem was solved completely and we have open free internet access here now
HUGE thanks to you and the developers behind v2ray

**Comment by Okelloraphael** [comment]
Help me with a working server setting
